### PR TITLE
Problem (#1483): client-rpc library and server dependencies are mixed

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- Seperate them, less dependencies in client-rpc library

### CODE REVIEW

File: client-rpc/src/lib.rs [hunk 1]
```diff
@@ -1,8 +1,21 @@
-mod program;
-/// Used by c api
+use std::fmt::Debug;
+
 pub mod rpc;
-mod server;
 
-pub fn run() {
-    crate::program::run_electron();
+pub fn to_rpc_error<E: ToString + Debug>(error: E) -> jsonrpc_core::Error {
```
Comment (devashishdxt) [inline client-rpc/src/lib.rs:5]: `Display` is more idiomatic?
```suggestion
pub fn to_rpc_error<E: Display + Debug>(error: E) -> jsonrpc_core::Error {
```

Comment (yihuang) [reply client-rpc/src/lib.rs:5]: Maybe use `Error` directly here is more proper?

Comment (devashishdxt) [reply client-rpc/src/lib.rs:5]: Yeah. But, some error types may not implement `Error`. You can try that. If it compiles fine, then it’s good.

Comment (yihuang) [reply client-rpc/src/lib.rs:5]: Then I think `ToString + Debug` is actually accurate here, `ToString` for the `to_string` method, `Debug` for the logging.

**Pull request merged**